trappy/ftrace.py

```diff
@@ -59,6 +59,21 @@ def _plot_freq_hists(allfreqs, what, axis, title):
                         r"(?P<timestamp>[0-9]+(?P<us>\.[0-9]+)?): (\w+:\s+)+(?P<data>.+)"
 )
 
+# This is just a helper to enable printing a helpful warning.
+# See GenericFTrace.__add_warnings
+class _UniqueNameWarner(object):
+    def __init__(self, owner, name, unique_word):
+        self.owner = owner
+        self.name = name
+        self.unique_word = unique_word
+
+    def __getattr__(self, attr):
```

Comment:
EDIT: this comment becomes void according to the next one, since that class can disappear altogether.
`__get__` could be used instead, so this object becomes a non-data descriptor and simply accessing it as an attribute will raise an exception:
```
# Currently that raises an exception:
owner.name.data_frame
# But that does not:
owner.name

# with __get__, this would raise an exception as well:
owner.name
```